### PR TITLE
Add heatmap for pairwise

### LINKED ISSUE #566: Add heatmap as an option for pairwise plot
@mirams suggests to bring back heatmap as an option for `pints.plot.pairwise`.

I cannot remember exactly why we moved away from it and use only scatter/kde plots, because I certainly remember I used to do heatmap instead of scatter/kde plots.

I think if there is no objection of adding it as an option, I will quickly do it with all other pints.plot tickets that I owe. @ben18785 @sanmitraghosh @MichaelClerx?

### PR DESCRIPTION
**Pull request by chonlei**
@mirams requested.

Close #566

### CODE REVIEW

File: pints/plot.py [hunk 1]
```diff
@@ -565,7 +566,10 @@ def pairwise(samples,
         is the number of parameters.
     ``kde``
         (Optional) Set to ``True`` to use kernel-density estimation for the
-        histograms and scatter plots.
+        histograms and scatter plots. Cannot use together with ``heatmap``.
+    ``heatmap``
+        (Optional) Set to ``True`` to plot heatmap for the pairwise plots.
+        Cannot use together with ``kde``.
     ``opacity``
```
Comment (MichaelClerx) [inline pints/plot.py]: Does opacity do anything for the heatmap?

Comment (chonlei) [reply pints/plot.py]: No, it should only affect the scatter plot -- neither kde or heatmap uses it.

**Comment by mirams** [comment]
Got an example of it?

**Comment by chonlei** [comment]
@mirams Not in the repo, but if you run `pints.plot.pairwise(samples, heatmap=True)`, you will get something like this: 

image

**Comment by chonlei** [comment]
@MichaelClerx Really not sure why it complained!! Looking at the log: 

image

It says `./pints/_mcmc/_dream.py` flake8 error, but I didn't change it!!

**Comment by chonlei** [comment]
It turned out to be a style error in master!! Fixed and rebased.

**Comment by chonlei** [comment]
Merge if Gary is happy (and pass tests).

**Comment by mirams** [comment]
Looks good.

File: pints/plot.py (hunk 1 above)
Comment (MichaelClerx) [reply pints/plot.py]: Please mention this in the docstring then!

**Pull request merged**